testnet/stacks-node/src/mockamoto.rs

```diff
@@ -475,75 +964,14 @@ impl MockamotoNode {
         Ok(block)
     }
 
-    fn mine_and_stage_block(&mut self) -> Result<(), ChainstateError> {
+    fn mine_and_stage_block(&mut self) -> Result<u64, ChainstateError> {
         let block = self.mine_stacks_block()?;
         let config = self.chainstate.config();
         let chainstate_tx = self.chainstate.db_tx_begin()?;
```

Comment:
This should be `self.chainstate.staging_db_tx_begin()`.  We'll ultimately be putting these blocks into a separate DB, so we should use the appropriate transaction initializer for it.

Reply:
Sure -- updated in 05274b6.